src/api/rasters.ts

```diff
@@ -0,0 +1,290 @@
+// Functions that call the rasters API
+// Centralized here so we know what is used and what isn't,
+// to start the Typescriptification process and to do error
+// handling in a more uniform way, if we do it.
+
+export interface OldRaster {
+  name: string;
+  organisation: string;
+  access_modifier: string;
+  observation_type: string;
+  description: string;
+  supplier: string;
+  supplier_code: string;
+  temporal: boolean;
+  interval?: number;
+  rescalable: boolean;
+  optiimizer: false;
+  aggregation_type: string;
+  options: string;
+  shared_with: string;
+}
+
+export interface OldRasterEdit {
+  name: string;
+  organisation: string;
+  access_modifier: string;
+  observation_type: string;
+  description: string;
+  supplier: string;
+  supplier_code: string;
+  aggregation_type: string;
+  options?: string;
+  shared_with: string;
+}
+
+export const fetchRasterV3 = async (uuid: string) => {
+  const response = await fetch(`/api/v3/rasters/${uuid}/`, {
+    credentials: "same-origin"
+  });
+
+  return response.json();
+};
+
+
+export const fetchRasterV4 = async (uuid: string, options: RequestInit = {
+  credentials: "same-origin"
+}) => {
+  const response = await fetch(`/api/v4/rasters/${uuid}/`, {
+    ...options,
+    method: "GET"
+  });
+
+  return response.json();
+};
+
+export const flushRaster = async (uuid: string) => {
```

Comment:
@caspervdw 
@remcogerlich 

If I understand correct flushing the raster now consists of these steps:
- set rasterlayer source to none
- delete raster source
- recreate raster source
- set rasterlayer source field to newly created raster source.
This is temporarily right? How will this in the final version work?

Also I am still not convinced of deleting data like this. 
Most software make it hard for the user to delete data. Your os and email has a trashbin. Google docs has history etc.
This is not only because user can do it by accident, but also because an attacker can do it. 
Remember any employee of organisation x can with few clicks delete all rasters for that organisation. 
Companies that take security serious will think twice before using this in a critical system.

Reply:
We use a trashbin on the backend, if you delete the RasterSource it will be hidden and only deleted after 14 days. A superuser can recover it from the admin panel. I hope that takes a way your worries.